src/sentry/services/hybrid_cloud/user/impl.py

```diff
@@ -64,7 +68,23 @@ def get_from_project(self, project_id: int) -> List[APIUser]:
         return self.get_many(project.member_set.values_list("user_id", flat=True))
 
     def get_by_actor_ids(self, *, actor_ids: List[int]) -> List[APIUser]:
-        return [UserService.serialize_user(u) for u in User.objects.filter(actor_id__in=actor_ids)]
+        return [
+            UserService.serialize_user(u)
+            for u in self.__base_user_query().filter(actor_id__in=actor_ids)
+        ]
 
     def close(self) -> None:
         pass
+
+    def __base_user_query(self) -> QuerySet:
+        return User.objects.select_related("avatar").extra(
```

Comment:
oh AWESOME I <3 this so much, you're a hero.  💪

Reply:
Glad you're okay with it. This felt messy to me, mostly because getting the django orm to do what I want _is_ messy. I explored a couple other options and this was the least bad 🤷 . _Something_ of this form is definitely required.